src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -66,6 +66,16 @@ public Expression Parse(string resourceType, string key, string value)
         }
 
         public IncludeExpression ParseInclude(string resourceType, string includeValue)
+        {
+           return (IncludeExpression)ParseIncludeOrRevinclude(resourceType, includeValue, false /* isReverse */);
+        }
+
+        public RevIncludeExpression ParseRevInclude(string resourceType, string includeValue)
+        {
+           return (RevIncludeExpression)ParseIncludeOrRevinclude(resourceType, includeValue, true /* isReverse */);
+        }
+
+        private IncludeBaseExpression ParseIncludeOrRevinclude(string resourceType, string includeValue, bool isReverse)
```

Comment:
IMO a cleaner implementation is to take a Func<IncludeExpression> in ParseIncludeOrRevinclude and avoid the switch by